### ISSUE TITLE
[FEAT] Public Leaderboard System

### ISSUE THREAD

**Issue by KendallDoesCoding** [KendallDoesCoding/quiz#33]
As of now the leaderboard/highscore only works locally (on the local computer), it would be amazing if the system can be website/server wide, but I think we may have to maintain a database or something for this.

**Comment by SrinjayKapri7** [comment]
we need database and deploy it in some server

**Comment by KendallDoesCoding** [comment]
> we need database and deploy it in some server

It cannot be deployed via a website?

**Comment by Srinja333** [comment]
yes but we need to deploy it heroku or others cloud platform to make your site workable for all

**Comment by SrinjayKapri7** [comment]
i think localstorage can be a good option

**Comment by KendallDoesCoding** [comment]
> i think localstorage can be a good option


Perfect! Can you work on this issue?

**Comment by SrinjayKapri7** [comment]
sure

**Comment by KendallDoesCoding** [comment]
assigned

**Comment by SrinjayKapri7** [comment]
you have already used local storage so we need to use databases, can i go for mongodb for that???

**Comment by KendallDoesCoding** [comment]
sure, go ahead!

*Thanks,*
*Kendall Does Coding*

*Portfolio
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fkendalldoescoding.gq&dID=1678861640513&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Portfolio>*
*Github
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fgithub.com%2FKendallDoesCoding&dID=1678861640513&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Github>*
*Youtube*
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fkendalldoescoding.gq%2FcodingYT&dID=1678861640513&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Youtube>

**Comment by KendallDoesCoding** [comment]
When will you make the PR @Srinja333?

**Comment by SrinjayKapri7** [comment]
soon

**Comment by SrinjayKapri7** [comment]
my localhost refuse to open how to run it?

**Comment by KendallDoesCoding** [comment]
> my localhost refuse to open how to run it?

any error message, screenshot? upload as a draft PR.

**Comment by SrinjayKapri7** [comment]
crbug/1173575, non-js module files deprecated.

**Comment by SrinjayKapri7** [comment]
i have just run it in vs code without debugging

**Comment by KendallDoesCoding** [comment]
it's ok make a draft PR let me see.

**Comment by SrinjayKapri7** [comment]
how to do that?

**Comment by KendallDoesCoding** [comment]
> how to do that?

you got git installed?

**Comment by SrinjayKapri7** [comment]
yes

**Comment by KendallDoesCoding** [comment]
Did you fork this repo to your profile and clone it using Git?

**Comment by SrinjayKapri7** [comment]
no sorry

**Comment by SrinjayKapri7** [comment]
i just clone from you

**Comment by KendallDoesCoding** [comment]
@Srinja333 - any update?

**Comment by SrinjayKapri7** [comment]
Work is ongoing.... I had exam so I missed some days.....

**Comment by KendallDoesCoding** [comment]
No worries. Thanks. Hope you did well in the exam!


*Thanks,*
*Kendall Does Coding*

*Portfolio
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fkendalldoescoding.gq&dID=1679807939406&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Portfolio>*
*Github
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fgithub.com%2FKendallDoesCoding&dID=1679807939406&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Github>*
*Youtube*
<https://cdc-aims.com/click?redirect=https%3A%2F%2Fkendalldoescoding.gq%2FcodingYT&dID=1679807939406&hashId=060de6478a47358f5315814fdd49f493dff92988d90dca1ba25e30911a8a60f18777e7&linkName=Youtube>

**Comment by Aditya1404Sal** [comment]
Hey ! does this issue / feature need working on ? , I'd be happy to work on it , i have worked with mongoDb before. Cheers !

**Comment by KendallDoesCoding** [comment]
> Hey ! does this issue / feature need working on ? , I'd be happy to work on it , i have worked with mongoDb before. Cheers !

Yes. Sure, you can work on this.

**Comment by KendallDoesCoding** [comment]
@Aditya1404Sal - I forgot to assign you to this issue, even though I commented.

Can you work on getting this done by next week, please? Because, I want to release this feature soon.

**Comment by Aditya1404Sal** [comment]
It will be done in a week

**Comment by ujju2000** [comment]
hey, this issue is resolved or i can work with you ?

**Comment by Aditya1404Sal** [comment]
@ujju2000 i had started the mongodb integration but it's taking up a considerable amount of time, so yeah you can takeover

**Comment by ujju2000** [comment]
okay, we just have to save the scores in the database ?

**Comment by RuchirDixit** [comment]
I believe we might need to use MongoDB Atlas in this case right? So that DB is publicly accessible?
Please let me know if there is any other way.

**Comment by KendallDoesCoding** [comment]
@ujju2000 @RuchirDixit - are both of you working on this?

**Comment by RuchirDixit** [comment]
I am just working on this issue for the sake of learning if @ujju2000 completes it he can raise a PR or else I will once it is solved.

**Comment by SlowMoschen** [comment]
Hi there! 
Is this Issue already resolved, if not maybe I can help with it.

If we want to save a Highscore of a User we need to save it to a Database. A MongoDB would be pretty suitable for it because it has a free storage option.

Setting up the Database and the corresponding express application is no big deal and I can create that quite quickly , but we need a host posibility for the backend so that the Database is always accessable for the Quizapp.

My suggestion for the MongoDB Highscore Schema would be:

**username** - a non unique username set by the user, how it is done already. (possibible duplicates, wich should be fine)
**highscore** -  the actual highscore reached by the user
**createdAt** - a timestamp when the Highscore was saved to the Database (we could display this then in the leaderboard)

To save and access the Highscores we should only need to swap the current localStorage logic with a POST/GET request to the express app.
For protection against a reuqets we dont want we can use cors and set the only allowed origin to the domain name.
